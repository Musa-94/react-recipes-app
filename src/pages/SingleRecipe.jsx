import React from 'react';

const SingleRecipe = () => {
    alert('SingleRecipe')
    return (
        <div>
            <h1> Hello Single Recipes Page </h1>
        </div>
    );
}

export default React.memo(SingleRecipe);
