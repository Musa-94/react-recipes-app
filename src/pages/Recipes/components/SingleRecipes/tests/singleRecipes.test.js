import React from 'react';
import SingleRecipe from '../SingleRecipe';

const props = {
    title: 'title',
    recipeId: '123',
    imageUrl: 'imageUrl',
    sourceUrl: 'sourceUrl',
    publisher: 'publisher',
};

describe('SingleRecipe snapshot', () => {
    it('should correct renderer', () => {
        const component = renderer(<SingleRecipe {...props}/>);

        expect(component).toMatchSnapshot();
    });
});