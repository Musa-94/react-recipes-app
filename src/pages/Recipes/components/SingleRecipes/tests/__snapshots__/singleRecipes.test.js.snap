// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleRecipe snapshot should correct renderer 1`] = `
<div
  className="col-1m mx-auto col-md-6 col-lg-4 my-3"
>
  <div
    className="card"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <img
      alt="img"
      className="img-card-top"
      id="123"
      src="imageUrl"
      style={
        Object {
          "height": "14rem",
        }
      }
    />
    <div
      className="card-body text-capitalize"
    >
      <h6>
        title
      </h6>
      <h6
        className="text-warning text-slanted"
      >
        provided by publisher
      </h6>
    </div>
    <div
      className="card-footer"
    >
      <a
        className="btn btn-primary text-capitalize"
        href="#/recipes/123"
      >
        details
      </a>
      <a
        className="btn btn-success mx-2 text-capitalize"
        href="sourceUrl"
        rel="noopener noreferrer"
        target="_blank"
      >
        recipe url
      </a>
    </div>
  </div>
</div>
`;
